packages/mcp/src/utils/registry-utils.ts

```diff
@@ -48,20 +56,41 @@ export async function registerToolsets(
 
   const toolsetsToEnable: Set<Toolset> = toolsets.includes('all')
     ? new Set(TOOLSETS)
+    // CORE toolset is always enabled
     : new Set([Toolset.CORE, ...(toolsets as Toolset[])]);
 
-  const newToolRegistry: Record<Toolset, McpTool[]> = await createToolRegistryFromProviders(
+  const toolsetRegistry: Record<Toolset, McpTool[]> = await createToolRegistryFromProviders(
     MCP_PROVIDER_REGISTRY,
     services
   );
 
+  ux.stderr('REGISTERING TOOLSETS (--toolsets)');
   for (const toolset of TOOLSETS) {
     if (toolsetsToEnable.has(toolset)) {
-      ux.stderr(`Registering tools from the '${toolset}' toolset.`);
+      ux.stderr(`Registering toolset: '${toolset}'`);
       // eslint-disable-next-line no-await-in-loop
-      await registerTools(newToolRegistry[toolset], server, useDynamicTools, allowNonGaTools);
+      await registerTools(toolsetRegistry[toolset], server, useDynamicTools, allowNonGaTools);
     } else {
-      ux.stderr(`Skipping registration of the tools from the '${toolset}' toolset.`);
+      ux.stderr(`   Skipping toolset: '${toolset}'`);
+    }
+  }
+
+  if (tools.length > 0) {
+    ux.stderr('REGISTERING TOOLS (--tools)');
```

Comment:
I _could_ have registered these tools to the "registering toolsets" loop above, but that made the startup logs even more jumbled than they already were. Instead it is checked here in its own loop.